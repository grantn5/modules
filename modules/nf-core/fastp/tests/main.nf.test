nextflow_process {

    name "Test Process FASTP"
    script "modules/nf-core/fastp/main.nf"
    process "FASTP"
    tag

    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                // define inputs of the process here. Example:
                // input[0] = file("test-file.txt")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
