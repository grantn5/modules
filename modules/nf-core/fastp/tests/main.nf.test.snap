{
    "test_fastp_paired_end_reads2_size": {
        "content": [
            396
        ],
        "timestamp": "2023-10-16T16:54:33.338157766"
    },
    "fastp test_fastp_interleaved_json": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": true
                    },
                    "test.fastp.json:md5,168f516f7bd4b7b6c32da7cba87299a4"
                ]
            ]
        ],
        "timestamp": "2023-10-16T16:54:50.688715226"
    },
    "test_fastp_single_end_json": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": true
                    },
                    "test.fastp.json:md5,c852d7a6dba5819e4ac8d9673bedcacc"
                ]
            ]
        ],
        "timestamp": "2023-10-16T16:54:21.346697558"
    },
    "test_fastp_single_end_trim_fail_reads": {
        "content": [
            [
                "@ERR5069949.2151832 NS500628:121:HK3MMAFX2:2:21208:10793:15304/1",
                "TCATAAACCAAAGCACTCACAGTGTCAACAATTTCAGCAGGACAACGCCGACAAGTTCCGAGGAACATGTCTGGACCTATAGTTTTCATAAGTCTACACACTGAATTGAAATATTCTGGTTCTAGTGTGCCCTTAGTTAGCAATGTGCGT"
            ]
        ],
        "timestamp": "2023-10-16T16:55:08.145072391"
    },
    "test_fastp_paired_end_reads2": {
        "content": [
            [
                "@ERR5069949.2151832 NS500628:121:HK3MMAFX2:2:21208:10793:15304/2",
                "ATGTGTACATTGGCGACCCTGCTCAATTACCTGCACCACGCACATTGCTAACTAAGGGCACACTAGAACCAGAATATTTCAATTCAGTGTGTAGACTTATGAAAACTATAGGTCCAGACATGTTCCTCGGAACTTGTCGGCGTTGTCCTG"
            ]
        ],
        "timestamp": "2023-10-16T16:54:33.321997927"
    },
    "test_fastp_paired_end_reads1": {
        "content": [
            [
                "@ERR5069949.2151832 NS500628:121:HK3MMAFX2:2:21208:10793:15304/1",
                "TCATAAACCAAAGCACTCACAGTGTCAACAATTTCAGCAGGACAACGCCGACAAGTTCCGAGGAACATGTCTGGACCTATAGTTTTCATAAGTCTACACACTGAATTGAAATATTCTGGTTCTAGTGTGCCCTTAGTTAGCAATGTGCGT"
            ]
        ],
        "timestamp": "2023-10-16T16:54:33.298896768"
    },
    "fastp test_fastp_interleaved_size": {
        "content": [
            792
        ],
        "timestamp": "2023-10-16T16:54:50.671505497"
    },
    "test_fastp_single_end_trim_fail_reads_size": {
        "content": [
            396
        ],
        "timestamp": "2023-10-16T16:55:08.16271281"
    },
    "fastp test_fastp_interleaved_reads": {
        "content": [
            [
                "@ERR5069949.2151832 NS500628:121:HK3MMAFX2:2:21208:10793:15304/1",
                "TCATAAACCAAAGCACTCACAGTGTCAACAATTTCAGCAGGACAACGCCGACAAGTTCCGAGGAACATGTCTGGACCTATAGTTTTCATAAGTCTACACACTGAATTGAAATATTCTGGTTCTAGTGTGCCCTTAGTTAGCAATGTGCGT"
            ]
        ],
        "timestamp": "2023-10-16T16:47:35.215031288"
    },
    "test_fastp_single_end_trim_fail_reads_fail": {
        "content": [
            [
                "@ERR5069949.885966 NS500628:121:HK3MMAFX2:4:11610:19682:20132/1 failed_quality_filter",
                "GTCTAATCATAATTTCTTGGTACAGGCTGGTATTGTTCATCTCAGGGTTATTGGACATTCTATGCAAAATTGTGTACTT"
            ]
        ],
        "timestamp": "2023-10-16T16:55:08.154298921"
    },
    "test_fastp_single_end_trim_fail_reads_fail_size": {
        "content": [
            4
        ],
        "timestamp": "2023-10-16T16:55:08.17061515"
    },
    "versions": {
        "content": [
            [
                "versions.yml:md5,48ffc994212fb1fc9f83a74fa69c9f02"
            ]
        ],
        "timestamp": "2023-10-16T16:47:35.413977197"
    },
    "test_fastp_single_end_reads": {
        "content": [
            [
                "@ERR5069949.2151832 NS500628:121:HK3MMAFX2:2:21208:10793:15304/1",
                "TCATAAACCAAAGCACTCACAGTGTCAACAATTTCAGCAGGACAACGCCGACAAGTTCCGAGGAACATGTCTGGACCTATAGTTTTCATAAGTCTACACACTGAATTGAAATATTCTGGTTCTAGTGTGCCCTTAGTTAGCAATGTGCGT"
            ]
        ],
        "timestamp": "2023-10-16T16:54:21.32164332"
    },
    "test_fastp_single_end_reads_size": {
        "content": [
            396
        ],
        "timestamp": "2023-10-16T16:54:21.330536959"
    },
    "test_fastp_paired_end_reads1_size": {
        "content": [
            396
        ],
        "timestamp": "2023-10-16T16:54:33.332775247"
    },
    "test_fastp_single_end_trim_fail_json": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": true
                    },
                    "test.fastp.json:md5,9a7ee180f000e8d00c7fb67f06293eb5"
                ]
            ]
        ],
        "timestamp": "2023-10-16T16:55:08.18137786"
    }
}